airflow/www/views.py

```diff
@@ -2550,7 +2550,10 @@ def set_task_instance_state(self, ids, target_state, session=None):
             count = len(ids)
             for id in ids:
                 task_id, dag_id, execution_date = id.split(',')
-                execution_date = datetime.strptime(execution_date, '%Y-%m-%d %H:%M:%S')
+                dt_format = ('%Y-%m-%d %H:%M:%S.%f'
```

Comment:
nit: looks like the execution_date string can be directly used in the ORM query without having to parse it into a datetime object, that could make your fix even simpler

Reply:
Thanks, @jgao54. I'll check it.

Reply:
@jgao54, sqlalchemy doesn't seem to be able to magically parse it.

Your idea is to use the string directly on the query filter, right?

Reply:
@boechat107 that's right! hmm that's odd. The query filter should be able to handle string directly like `TI.execution_date == '2017-10-25 10:10:10.100'`. It works for me when I tested it just now.

What also leads me to this conclusion is looking at [these lines](https://github.com/boechat107/incubator-airflow/blob/bdfda2e7c0771aa8f07da2a714e9eb38921d0d52/airflow/www/views.py#L2525-L2529) we are already doing this with action_clear.

Reply:
Hum, it makes sense. I'll test this **action_clear** the same way I tested **set_task_instance_state**.

Reply:
I replicated the test for **action_clear** and the code seems to fail in the same way. I didn't experience this failure in production because I don't remember of having the **clear** option (older version?).

Or my local virtualenv has something wrong or **action_clear** also needs to be patched.

Reply:
@boechat107 yeah the `clear` option is relatively new. I can't replicate this issue though, both action_clear and  set_task_instance_state work for me via the UI. I tried locally in both PY2/3 environments with mysql db, curious if you use a different db?

Reply:
@jgao54, did you try to set the state of tasks created from manually triggered DAGs?

The steps I used to build my local environment are:

* virtualven: `virtualenv -p /usr/bin/python3 venv` (which gave me Python 3.5.4)
* the dependencies: `pip install -e .[devel,hive]` (`devel` was not enough)
* installation of some OS libraries, specially for MySQL
* different values for `AIRFLOW_HOME` and `AIRFLOW_DB` in **run_unit_tests.sh**

Reply:
I created another branch to apply my tests to the original code: https://travis-ci.org/boechat107/incubator-airflow/builds/292832358

I think these results can give us more useful data to discuss.

Reply:
The original code doesn't pass the new tests, throwing the same error I got when using the UI.

Reply:
@boechat107 
ah I was finally able to replicate your bug. Typically execution date is rounded to the nearest second (i.e. see [this line](https://github.com/apache/incubator-airflow/blob/master/airflow/api/common/experimental/trigger_dag.py#L35)), that's why I couldn't regenerate that bug at first when I ran a command like `airflow trigger_dag -e `2017-01-01 00:00:00.123`, because it would just round the microseconds to 0.

One thing that's worth noting is that `ValueError: unconverted data remains: ..372649` seem to suggests that the execution_date string looks like `'%Y-%m-%d %H:%M:%S..372649' (2 periods in the date string). Seems like a bug coming from elsewhere. 

That aside, it seems reasonable to be able to support microseconds precision, thanks for looking into this!

Reply:
Yes, true. This exception happened to me many many times, but I confess that usually I use the command line only to run `airflow test ...`. When I want to manually trigger a DAG, I use the UI and I thought there was some intention in keeping the fractional part.

Thank you very much for your attention, @jgao54! I hope we can fix this.